ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/PeerSelection/Cardano/PublicRootPeers.hs

```diff
@@ -58,92 +63,96 @@ arbitraryPublicRootPeers peeraddrs = do
 
   return (PublicRootPeers.fromMapAndSet publicConfigPeersMap bootstrapPeers normalLedgerPeers bigLedgerPeers)
 
-instance (Arbitrary peeraddr, Ord peeraddr) =>
-         Arbitrary (PublicRootPeers peeraddr) where
-    arbitrary = do
-        peeraddrs <- arbitrary
-        arbitraryPublicRootPeers peeraddrs
+instance ( Arbitrary peeraddr
+         , Ord peeraddr
+         , extraPeers ~ CardanoExtraPeers peeraddr
```

Comment:
why not replace `extraPeers` in the instance head.

Reply:
To avoid having to use `FlexibleInstances`, but I have made this change in another module and included it, so I will do it in this one as well

Reply:
`FlexibleInstances` will be on by default, once we switch to `GHC202X`.